### DISCUSSION TITLE
Move to ROCm

### DISCUSSION THREAD

**Discussion by hkpD** [lshqqytiger/stable-diffusion-webui-directml#277] [General]
Hi guys,

I kept having constant issues with this project on Windows 11. I would have to rollback changes and do "git checkout xxxx" which would ruin the whole directory structure and cause even more problems.

So I installed Manjaro Linux and went to the Automatic1111 wiki and followed the "Arch Linux" installation section. 

I went from having 1.60 it/s on Windows 11 to having 6.80 it/s on Manjaro Linux. It's not even comparable. If you are serious about this stuff, I highly recommend you move to ROCm. You will not regret it. 

Basically, I'm not that technically advanced and I was able to do it. All you have to do is install Manjaro Linux and then follow the "Arch Linux" section in the Automatic1111 wiki.

**Comment by pw405** [comment]
Interesting, I was thinking about this... but knowing my luck I would probably get it all setup and then ROCm for Windows would come out next day!   So with Linux version, do you have full compatibility identical to a CUDA/Nvidia system?  I know many extensions, like Dream Booth, didn't work with DirectML version.

**Reply by hkpD** [reply]
Yes, it has full functionality. I'm running the original Automatic1111 so it has every single feature that is listed on the Automatic1111 page. 

Are you able to set some disk space aside and partition it? Then you can install Linux Manjaro on the side and stil lkeep your Windows install. 

I used to be so anti-Linux, but I swear, I'm starting to prefer Linux Manjaro over Windows... Windows is only good for gaming. The rest is done so much better by Linux

**Reply by pw405** [reply]
Any helpful guides you used to set up Linux correctly? I have plenty of disk space atm, thinking about giving it a shot.  I understand dual boot on current OS drive is best.

**Reply by hkpD** [reply]
Yep, I just did it again yesterday with EndeavourOS. All I did was install the operating system and then followed this guide at the bottom for Arch. https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-AMD-GPUs#install-on-amd-and-arch-linux

**Comment by MedDeatH** [comment]
while it might get better for rdna gpus, rx 580(which i tested one) gets same,or even worse (in hires fix) performance.only thing better is gpu memory management

**Comment by Flameancer** [comment]
He's right, got a 7800xt and first dual booted a ubuntu build and got it working (Idk how @hkpD did it on manjaro, it would never launch on a fresh new build for me). Not running it right now but the times with ROCm are drastically faster than DirectML. I'm getting around 1.37s/it on Windows + DirectML. I will try manjaro again and edit back.

**Reply by hkpD** [reply]
I booted up my EndeavourOS install (which is kind of the same thing as Manjaro) and tried to run sd-webui but it wouldn’t launch like last time probably because the Python version got upgraded to 3.11. I had to manually install python3.10, create a python3.10 venv, and installed requirements.txt. Then I had to pip uninstall torch and torchvision in the venv and reinstalled torch using the ROCM command from PyTorch.org.

**Reply by Flameancer** [reply]
Couldn't get it quit working on Manjaro but I got the UI to launch which is further than last time. But on Ubuntu 22.04 I get like 11-13s/it on ROCm5.7. DirectML is good if you wanna get your feet wet on AMD. But I would recommend dual booting Linux to get faster results.

**Reply by StarlitLuna** [reply]
How are you getting 11-13it/s??

I have a 7900 XTX, installed on ubuntu and finally got it working with pytorch using rocm5.7. Detects my GPU and everything but I get around 2.5it/s tops...

**Reply by Flameancer** [reply]
It depends on my settings. When I originally had commented I was just really getting into stable diffusion and didn't do many setting changes. Last week I was messing around with changing settings along with properly using high rez fix and my it/s were closer to that 2-3. https://vladmandic.github.io/sd-extension-system-info/pages/benchmark.html

**Reply by StarlitLuna** [reply]
I haven't touched my settings at all and I'm getting 1.5 - 3 it/s...

**Reply by Flameancer** [reply]
> How are you getting 11-13it/s??
> 
> 
> 
> I have a 7900 XTX, installed on ubuntu and finally got it working with pytorch using rocm5.7. Detects my GPU and everything but I get around 2.5it/s tops...

I just noticed you said you're using 5.7. I was running the nightly version of PyTorch with 6.0. I'm on my honeymoon now and I had PC trouble before I left so I won't be able to test until I get back and reinstall after Christmas but I saw that 6.0 is officially out so try running 6.0 and see if you get better results.

**Comment by ghost** [comment]
Rocm is miles better than Directml with my 5700xt. Windows 11 512x768 Euler a 50 steps 2mins, ROCm 29 sec. Linux mint 21.2 Cinnamon "EDGE" Edition is more stable than Ubuntu 22.04. Try that instead. I'm able to gen 1920x1080 without tilted VAE in Mint with SDXL.

**Comment by binarydepth** [comment]
ROCn has OpenCL runtime that can be used to run on old GPU's. Just chiming in that bit
